sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -360,6 +360,33 @@ def delete_share(
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def undelete_share(self, deleted_share_name, deleted_share_version, **kwargs):
```

Comment:
`deleted_share_name` shouldn't be a parameter - as the `ShareClient` already has a name.
If we want to pass in a name to this function - then we should move this function up to the parent client.

Reply:
I see we're checking the name against `self.share_name` below - in that case we should definitely remove this parameter and just use `self.share_name` directly.

Reply:
deleted_share_name is the name when it is deleted, the name on ShareClient is the one you want to restore to. Currently there's a bug in service so that deleted_share_name should be the same as ShareClient.share_name, while in stg74 they could be different